tests/lib/rules/no-implicit-coercion.js

```diff
@@ -46,6 +51,15 @@ ruleTester.run("no-implicit-coercion", rule, {
         {code: "+foo.bar", errors: [{message: "use `Number(foo.bar)` instead.", type: "UnaryExpression"}]},
         {code: "1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
         {code: "foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*bar*1*foo", errors: [
+            {message: "use `Number(bar)` instead.", type: "BinaryExpression"},
+            {message: "use `Number(foo)` instead.", type: "BinaryExpression"}
+        ]},
+        {code: "1234*1*foo*bar", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
```

Comment:
These seem wrong to me. Multiplying by 1 might be unnecessary, but that's not really the point of this rule.